protobuf_release.bzl

```diff
@@ -0,0 +1,34 @@
+"""
+Generates package naming variables for use with rules_pkg.
+"""
+
+load("@rules_pkg//:providers.bzl", "PackageVariablesInfo")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load(":protobuf_version.bzl", "PROTOBUF_VERSION")
+
+def _package_naming_impl(ctx):
+  values = {}
+  values["product_name"] = ctx.attr.product_name
+  values["version"] = PROTOBUF_VERSION
+
+  # infer from the current cpp toolchain.
+  toolchain = find_cpp_toolchain(ctx)
+  values["cpu"] = toolchain.cpu
+  
+  return PackageVariablesInfo(values = values)
+
+
+package_naming = rule(
+  implementation = _package_naming_impl,
+    attrs = {
+      "product_name": attr.string(
+          doc = "Placeholder for our final product name.",
+      ),
```

Comment:
Why does this need to be a variable? What other product names would we make other than protoc?

Reply:
this could be used for other runtimes etc, but I think we can just hardcode this in place for now.